server/lib/pencil_space_server_web/controllers/game_controller.ex

```diff
@@ -4,33 +4,47 @@ defmodule PencilSpaceServerWeb.GameController do
   """
 
   use PencilSpaceServerWeb, :controller
-  alias PencilSpaceServer.{GameMonitor}
+  alias PencilSpaceServer.{Game}
+  import PencilSpaceServerWeb.ControllerHelpers
 
-  def create(conn, _params) do
-    game_id = Nanoid.generate()
+  def create(
+        conn,
+        %{
+          "host" => %{
+            "id" => _id,
+            "name" => _name,
+            "avatar" => _avatar
+          }
+        } = host
+      ) do
 
-    case GameMonitor.start(game_id) do
-      {:ok, _pid} ->
-        conn
-        |> put_status(200)
-        |> json(%{id: game_id})
+    case Game.start do
+      {:ok, name} ->
+        Game.update(name, host)
+        render_json(conn, :created, %{name: name})
       {:error} ->
-        conn
-        |> put_status(503)
-        |> json(%{error: "Could not create a new game"})
+        render_json(conn, :service_unavailable, %{error: "Could not create a new game"})
     end
   end
 
-  def join(conn, %{"id" => game_id}) do
-    case GameMonitor.exists?(game_id) do
+  def join(
+        conn,
+        %{
+          "name" => name,
+          "participant" => %{
+            "id" => _id,
+            "name" => _name,
+            "avatar" => _avatar
+          }
+        } = params
+      ) do
+
+    case Game.presence(name) do
       {:ok, :exists} ->
-        conn
-        |> put_status(200)
-        |> json(%{id: game_id})
+        Game.update(name, Map.take(params, ["participant"]))
+        render_json(conn, :accepted, %{name: name})
```

Comment:
The `Game.update` will invoke a `handle_call` callback in `Monitor, which is sync. If sufficiently large number of people join the same group at the same time, the laggards will see high latency in http response. Maybe we can change the participant list update to be async by moving it to handle_cast.

This again is alright, I am just trying to highlight the scenarios in which the Pencil server may run into trouble.